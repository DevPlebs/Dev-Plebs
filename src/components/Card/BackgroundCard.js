import { string, bool } from "prop-types"
import styled from "@emotion/styled"
import tw from "tailwind.macro"

import placeholderBg from "../../images/hills.svg"
import Card from "./Card"

export const BackgroundCard = styled(Card)`
  ${tw`text-white text-shadow text-center no-underline`}
  text-shadow: 2px 2px 2px rgba(0, 0, 0, 0.3);
  background: linear-gradient(rgba(0, 0, 0, 0.35), rgba(0, 0, 0, 0.35)),
    url(${props => props.backgroundImage});
  background-size: cover;
  background-position-x: center;
  background-position-y: 70%;
  ${props =>
    props.hover &&
    `:hover {
    background: linear-gradient(rgba(0, 0, 0, 0.25), rgba(0, 0, 0, 0.25)),url(${props.backgroundImage});
    background-size: cover;
    background-position-x: center;
    background-position-y: 55%;
  }`}
`

BackgroundCard.propTypes = {
  backgroundImage: string,
  hover: bool,
}

BackgroundCard.defaultProps = {
  backgroundImage: placeholderBg,
  hover: true,
}
export default BackgroundCard
