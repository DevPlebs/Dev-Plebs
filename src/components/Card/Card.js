import React from "react"
import { node, string, oneOfType, object } from "prop-types"

import {
  CardContainer,
  DefaultCard,
  CardBody,
  CardHeader,
  CardTitle,
  CardDeck,
} from "./styles"

const Card = ({ className, header, title, text, deck, as, to, children }) => {
  return (
    <CardContainer className={className}>
      <DefaultCard className="DefaultCard" as={as} to={to}>
        {header && <CardHeader>{header}</CardHeader>}
        <CardBody>
          {title && <CardTitle>{title}</CardTitle>}
          {text && <p>{text}</p>}
          {children}
        </CardBody>
      </DefaultCard>
      {deck && <CardDeck>{deck}</CardDeck>}
    </CardContainer>
  )
}

Card.propTypes = {
  className: string,
  header: string,
  title: string,
  text: string,
  deck: string,
  as: oneOfType([string, object]),
  to: string,
  children: node,
}

Card.defaultProps = {
  header: "",
  className: "",
  as: "",
  to: "",
  title: "",
  text: "",
  deck: "",
}

export default Card
