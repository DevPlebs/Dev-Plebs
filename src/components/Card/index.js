export { default as Card } from "./Card"
export { BackgroundCard } from "./BackgroundCard"
export { DefaultCard, CardContainer, CardDeck } from "./styles"
