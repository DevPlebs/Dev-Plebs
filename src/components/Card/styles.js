import styled from "@emotion/styled"
import tw from "tailwind.macro"

// Core Card styles
export const DefaultCard = styled.div`
  ${tw`flex flex-wrap shadow bg-white rounded-lg shadow hover:shadow-lg`}
  width: 100%;
  position: relative;
  font-family: montserrat;
  transition: 0.35s;
  :hover {
    transform: ${props => props.hover && "translateY(-10px)"};
  }
`

export const CardHeader = styled.h4`
  ${tw`py-2 pl-2 pr-6 mb-0 text-white rounded-tl-lg`}
  max-height: 2rem;
  font-family: Rubik;
  background-color: #c04abc;
  clip-path: polygon(0 0, 100% 0%, 85% 100%, 0 100%);
`

export const CardBody = styled.div`
  ${tw`p-4 w-full text-lg flex flex-col`}
  h1, h2, h3, h4, h5 {
    font-family: Rubik;
  }
`

export const CardContainer = styled.div`
  ${tw`relative flex flex-row flex-wrap justify-center md:flex-no-wrap`}
`
export const CardText = styled.p`
  color: #000;
`
export default DefaultCard
