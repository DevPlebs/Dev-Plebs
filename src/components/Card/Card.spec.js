import React from "react"
import { create } from "react-test-renderer"

import Card from "./Card"

describe("Card", () => {
  it("renders correctly", () => {
    const tree = create(<Card />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
