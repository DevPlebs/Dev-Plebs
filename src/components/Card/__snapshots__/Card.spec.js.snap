// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
.emotion-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:768px) {
  .emotion-2 {
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.1),0 1px 2px 0 rgba(0,0,0,0.06);
  background-color: #fff;
  position: relative;
  font-family: montserrat;
  -webkit-transition: 0.35s;
  transition: 0.35s;
}

.emotion-0 {
  padding: 1rem;
  width: 100%;
  font-size: 1.125rem;
}

.emotion-0 h1,
.emotion-0 h2,
.emotion-0 h3,
.emotion-0 h4,
.emotion-0 h5 {
  font-size: 1.5rem;
  font-family: Rubik;
}

<div
  className=" emotion-2"
>
  <div
    className="DefaultCard emotion-1"
    to=""
  >
    
    <div
      className="emotion-0"
    >
      
      
    </div>
  </div>
  
</div>
`;
